tsdb/wal/watcher.go

```diff
@@ -252,10 +255,13 @@ func (w *Watcher) Run() error {
 		}
 
 		// For testing: stop when you hit a specific segment.
+
 		if currentSegment == w.MaxSegment {
 			return nil
 		}
 
+		// Sets the current Segment files name
+		w.SegmentFile = SegmentName("", currentSegment)
```

Comment:
Would it be possible for the watcher to have progressed to a new segment, without sending all samples from the previous segment? I think if you had a large enough buffer capacity in the queues that it could happen.

Reply:
Yes it is possible